api/tasks.go

```diff
@@ -362,6 +362,21 @@ func (m *MigrateStrategy) Copy() *MigrateStrategy {
 	return nm
 }
 
+type Volume struct {
```

Comment:
I'd urge us to have better code documentation for the significant data structures in this PR before merging, for our own sanity when we approach this code.  This is a follow up from https://github.com/hashicorp/nomad/pull/5923#discussion_r309283098

Reply:
@notnoop The structs were documented in `nomad/structs/volumes.go` already, I've copied the documentation across to here, but our api structs documentation is generally pretty nonexistent.